File: app/src/main/java/com/minar/randomix/fragments/RateBottomSheet.java
package com.minar.randomix.fragments;

import android.content.Intent;
import android.content.SharedPreferences;
import android.graphics.drawable.Animatable2;
import android.graphics.drawable.Drawable;
import android.net.Uri;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;

import androidx.annotation.Nullable;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.google.android.material.checkbox.MaterialCheckBox;
import com.minar.randomix.R;
import com.minar.randomix.activities.MainActivity;

public class RateBottomSheet extends BottomSheetDialogFragment {
    private final MainActivity activity;
    private final SharedPreferences.Editor editor;

    public RateBottomSheet(MainActivity activity, SharedPreferences.Editor editor) {
        this.activity = activity;
        this.editor = editor;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        // Inflate the bottom sheet, initialize the shared preferences and the recent options list
        View v = inflater.inflate(R.layout.rate_bottom_sheet, container, false);

        final String DO_NOT_SHOW_AGAIN = "do_not_show_again";

        // Animate the drawable in loop
        ImageView noRecentImage = v.findViewById(R.id.rateImage);
        Drawable animatedNoRecent = noRecentImage.getDrawable();
        if (animatedNoRecent instanceof Animatable2) {
            ((Animatable2) animatedNoRecent).registerAnimationCallback(new Animatable2.AnimationCallback() {
                @Override
                public void onAnimationEnd(Drawable drawable) {

                    ((Animatable2) animatedNoRecent).start();
                }
            });
            ((Animatable2) animatedNoRecent).start();
        }

        Button positiveButton = v.findViewById(R.id.positiveButton);
        Button negativeButton = v.findViewById(R.id.negativeButton);
        MaterialCheckBox checkBox = v.findViewById(R.id.neverAgainCheckbox);

        // Handling the positive button
        positiveButton.setOnClickListener(view -> {
            activity.startActivity(
                    new Intent(
                            Intent.ACTION_VIEW,
                            Uri.parse("market://details?id=" + activity.getPackageName())
                    )
            );
            editor.putBoolean(DO_NOT_SHOW_AGAIN, true);
            editor.commit();
            this.dismiss();
        });

        // Handling the negative button
        negativeButton.setOnClickListener(view -> this.dismiss());

        // Handling the "don't ask again" checkbox
        checkBox.setOnCheckedChangeListener((compoundButton, b) -> {
            editor.putBoolean(DO_NOT_SHOW_AGAIN, b);
            editor.commit();
        });

        return v;
    }
}

File: app/src/main/java/com/minar/randomix/fragments/RateBottomSheet.kt
package com.minar.randomix.fragments

import android.content.Intent
import android.content.SharedPreferences
import android.graphics.drawable.Animatable2
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import androidx.core.net.toUri
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.google.android.material.checkbox.MaterialCheckBox
import com.minar.randomix.R
import com.minar.randomix.activities.MainActivity

class RateBottomSheet(
    private val activity: MainActivity,
    private val editor: SharedPreferences.Editor
) : BottomSheetDialogFragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        val v = inflater.inflate(R.layout.rate_bottom_sheet, container, false)
        val doNotShowAgain = "do_not_show_again"

        val noRecentImage = v.findViewById<ImageView>(R.id.rateImage)
        val animatedNoRecent = noRecentImage.drawable
        if (animatedNoRecent is Animatable2) {
            animatedNoRecent.registerAnimationCallback(object : Animatable2.AnimationCallback() {
                override fun onAnimationEnd(drawable: Drawable) { animatedNoRecent.start() }
            })
            animatedNoRecent.start()
        }

        v.findViewById<Button>(R.id.positiveButton).setOnClickListener {
            activity.startActivity(
                Intent(Intent.ACTION_VIEW, "market://details?id=${activity.packageName}".toUri())
            )
            editor.putBoolean(doNotShowAgain, true).commit()
            dismiss()
        }
        v.findViewById<Button>(R.id.negativeButton).setOnClickListener { dismiss() }
        v.findViewById<MaterialCheckBox>(R.id.neverAgainCheckbox).setOnCheckedChangeListener { _, b ->
            editor.putBoolean(doNotShowAgain, b).commit()
        }
        return v
    }
}
